### ISSUE TITLE
Refresh button in TOC give "An unexpected error has occurred "

### ISSUE THREAD

**Issue by A-Nicoladie** [stevencohn/OneMore#1674]
# Problem to Solve

When I click on th refresh button in TOC, I have the following message : 

> Une erreur inattendue s'est produite.

Or probably in english : `An unexpected error has occurred`

# To Reproduce

1. Add some titles
2. Add a TOC
3. Click on refresh

# Expected Behavior

A refresh of the TOC

# Screenshots

![image](https://github.com/user-attachments/assets/b433c66d-926f-4f34-a44b-ea484c23d97b)

# Environment

   One More (

 - OneNote Version: on Windows 11 (23H2) with O365 64bits / OneNote 16.0.18025.20214
 - OneMore Version 6.6.1

# Additional Context

Nothing found in the log file :
[OneMore.log](https://github.com/user-attachments/files/17800236/OneMore.log)

THE FINE PRINT This is a hobby project that I started for myself. I'm happy to take requests and will promise to
consider each one carefully but offer no guarantees that I will ultimately agree to their
usefulness or have time to implement any of them. If OneNote offers a "close enough" work-around
then I will probably reject the request - you've been warned!

**Comment by stevencohn** [comment]
Unable to reproduce the simple example shown above. Are you sure you have the latest OneMore.log? It doesn't even include the line showing the refresh was executed??

**Comment by A-Nicoladie** [comment]
Yes this is the last. No matter how many times I click on refresh, nothing is written in log about it.


The insert is logged:

```
01|03:25:52.844| ..ready
14|03:25:52.848| hashtag service waiting, Ready
03|03:26:02.187| hashtag SERVICE scanned 718 pages, 6/6 notebooks, 165 sections, updating 0 pages, saving 0 tags, in 6325ms
03|03:26:02.328| Running command InsertTocCommand
10|03:26:06.683| ..build toc for page 
10|03:28:07.353| hashtag SERVICE scanned 718 pages, 6/6 notebooks, 165 sections, updating 0 pages, saving 0 tags, in 5107ms
```

**Comment by stevencohn** [comment]
If you hover your mouse cursor over the Refresh link, does it look like this?

<img width="259" alt="image" src="https://github.com/user-attachments/assets/2a7cd05b-9cb7-4080-b3d2-9e2743e24fed">


Try deleting the Table of Contents and try recreating it.

**Comment by A-Nicoladie** [comment]
Same problem after uninstall/install

I recreated a page with titles, added a TOC, and clicked on refresh.

Here's the log :  [OneMore.log](https://github.com/user-attachments/files/17816858/OneMore.log)

**Comment by stevencohn** [comment]
Your log isn't showing that the refresh link is clicked, which still indicates that the `onemore:` protocol isn't registered.

If you can, [download the iq.ps1 file](https://github.com/stevencohn/OneMore/blob/main/iq.ps1), and run it from an administrative PowerShell console. It makes no changes! It only reports configuration. It should show output like this, with the `onemore:` protocol highlighted below:

<img width="804" alt="image" src="https://github.com/user-attachments/assets/ac793e47-9f89-4d9a-9d2f-54a42f5e28f2">

**Comment by stevencohn** [comment]
Do you still have the installation log that you can attach here? `%temp%\OneMoreSetup.log`

**Comment by A-Nicoladie** [comment]
Result of PowerShell script

```
Versions...
OK Office version is 16.0
OK OneNote version is 15.0 (64-bit)

AppID...
OK Registry::HKEY_CLASSES_ROOT\AppID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}

Root...
OK Registry::HKEY_CLASSES_ROOT\onemore

Shell...
OK Registry::HKEY_CLASSES_ROOT\onemore\shell\open\command
= "C:\Program Files\River\OneMoreAddIn\OneMoreProtocolHandler.exe" %1 %2 %3 %4 %5

AddIn...
OK Registry::HKEY_CLASSES_ROOT\River.OneMoreAddIn
OK Registry::HKEY_CLASSES_ROOT\River.OneMoreAddIn.1

CLSID...
OK Registry::HKEY_CLASSES_ROOT\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}
OK Registry::HKEY_CLASSES_ROOT\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\InprocServer32
OK Registry::HKEY_CLASSES_ROOT\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\InprocServer32\6.6.1.0
= Assembly = River.OneMoreAddIn, Version=6.6.1.0, Culture=neutral, PublicKeyToken=null
= CodeBase = C:\Program Files\River\OneMoreAddIn\River.OneMoreAddIn.dll
= RuntimeVersion = v4.0.30319
= Class = River.OneMoreAddIn.AddIn
OK Registry::HKEY_CLASSES_ROOT\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\ProgID
OK Registry::HKEY_CLASSES_ROOT\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\Programmable
OK Registry::HKEY_CLASSES_ROOT\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\TypeLib
OK Registry::HKEY_CLASSES_ROOT\CLSID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}\VersionIndependentProgID

User...
OK Registry::HKEY_CURRENT_USER\SOFTWARE\Classes\AppID\{88AB88AB-CDFB-4C68-9C3A-F10B75A5BC61}
OK Registry::HKEY_CURRENT_USER\SOFTWARE\Microsoft\Office\OneNote\AddIns\River.OneMoreAddIn
OK Registry::HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\River.OneMoreAddIn.dll
= C:\Program Files\River\OneMoreAddIn\River.OneMoreAddIn.dll
OK Registry::HKEY_CURRENT_USER\SOFTWARE\Policies\Microsoft\Office\16.0\Common\Security\Trusted Protocols\All Applications\onemore:

WebView2...
OK Registry::HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\EdgeUpdate\Clients\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}
OK location = C:\Program Files (x86)\Microsoft\EdgeWebView\Application
= version = 130.0.2849.80
= location = C:\Program Files (x86)\Microsoft\EdgeWebView\Application
```


There is no `%temp%\OneMoreSetup.log` file (even after reinstalling the application again)

**Comment by A-Nicoladie** [comment]
The file contains only these lines

> pipe client connected
> writing 'onemore://InsertTocCommand/refresh/style0'
> 

It's a few days old.

**Comment by stevencohn** [comment]
Well that's good! Those two lines should be written every time you click refresh. So we know the onemore: protocol is being registered correctly and invoked.

Now to that debug version....

**Comment by A-Nicoladie** [comment]
Sorry for this long silence. Just to say:
- that these lines appear only once in the file, despite having clicked several times on the button
- the file's last update date is earlier than the date of the last reinstallation (file update date : 2024-11-14 // last fresh installation I do : 2024-11-21)

**Comment by A-Nicoladie** [comment]
Hi !
It seems that version 6.7.2 fixes this problem 👍

Thanks !
